src/units_for_slot_test.rs

```diff
@@ -118,7 +129,9 @@ mod units_for_slot_tests {
         }
     }
 
-    fn get_expected_response(channel: Channel, units: &[AdUnit]) -> UnitsForSlotResponse {
+    fn get_expected_response<'a>(
+        channels_with_units: Vec<ChannelUnitsPair<'a>>,
```

Comment:
a dedicated struct is not required since it adds some boilerplate.

I think we can we use crete the campaigns with `get_mock_campaign` directly and pass the `Vec<ResponseCampaign>` to `get_expected_response`